### PR TITLE
[wip][playground][refactor]: refactor the playground render piece for reuse

### PR DESCRIPTION
**Pull request by legomushroom**
[wip] The PR refactors the code around the rendering of the playground preview with intent to reuse it in the preview server. We need this server for security boundaries in VSO. (and potentially the ability to embed playgrounds or host them e.g. `https://gispad.com/lostintangent/{id}` which should open interesting use case scenarios).

Had to unwrap some circular dependencies so the PR becomes slightly bigger than intended. After refactoring we have the `renderPlaygroundHtml` class that accepts a `gist` and renders the resulting HTML as a string(including adding libraries and transpile code if needed).

### CODE REVIEW

File: src/commands/addPlaygroundLibraryCommand.ts [hunk 1]
```diff
@@ -108,7 +109,10 @@ async function addDependencyLink(
   playgroundJSON[libraryType] = [...new Set(playgroundJSON[libraryType])];
 
   const updatedContent = JSON.stringify(playgroundJSON, null, 2);
-  vscode.workspace.fs.writeFile(uri, stringToByteArray(updatedContent));
+  (vscode.workspace as any).fs.writeFile(
```
Comment (lostintangent) [inline src/commands/addPlaygroundLibraryCommand.ts]: Why is this needed?

File: src/utils.ts [hunk 2]
```diff
@@ -14,10 +14,6 @@ import { FS_SCHEME, PLAYGROUND_FILE } from "./constants";
 import { Gist, SortOrder, store } from "./store";
 import { getGist } from "./store/actions";
 
-export function byteArrayToString(value: Uint8Array) {
```
Comment (lostintangent) [inline src/utils.ts]: I'd prefer to leave this here.

File: src/byteArrayToString.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export function byteArrayToString(value: Uint8Array) {
```
Comment (arjun27) [inline src/byteArrayToString.ts]: it feels a bit odd that this is in a separate file whereas `stringToByteArray` is in utils

File: src/commands/playground.ts [hunk 4]
```diff
@@ -567,16 +342,16 @@ async function newPlaygroundInternal(
 }
 
 export async function openPlayground(gist: Gist) {
-  const markupFile = getGistFileOfType(gist, PlaygroundFileType.markup);
-  const stylesheetFile = getGistFileOfType(gist, PlaygroundFileType.stylesheet);
-  const scriptFile = getGistFileOfType(gist, PlaygroundFileType.script);
+  console.log(JSON.stringify(gist, null, 2));
```
Comment (arjun27) [inline src/commands/playground.ts]: remove log?

**Comment by lostintangent** [comment]
Closing this PR for now, since the Code has changed significantly since it was first submitted.
